state_machine/drivers/emulation/lib/radio_driver.py

```diff
@@ -11,22 +11,51 @@ def __init__(self, data, probability=1.0):
 
     def observe(self):
         """If the packet is sucessfully received, return the data, otherwise return None"""
-        if random.random() < self.probability:
+        if random.random() <= self.probability:
             return self.data
         else:
             return None
 
+class RadioDebug:
+
+    def __init__(self, radio):
+        self.radio = radio
+        self.last_tx_packet = None
+
+    def push_rx_queue(self, packet):
+        """Debug function to push a packet into the rx queue (fifo)"""
+        self.radio._rx_queue.put(packet)
+
+    def reset(self):
+        self.clear_rx_queue()
+        self.clear_tx_queue()
+
+    def clear_tx_queue(self):
+        """Debug function to clear the tx queue"""
+        self.radio._tx_queue = queue.Queue()
+
+    def clear_rx_queue(self):
+        """Debug function to clear the rx queue"""
+        self.radio._rx_queue = queue.Queue()
+
 class Radio:
     def __init__(self):
         self.node = 0
         self.listening = False
 
-        self._rx_queue = queue.LifoQueue()
+        self._rx_queue = queue.Queue()
         self._rx_time_bias = 0.5
         self._rx_time_dev = 0.3
+
+        self._tx_queue = queue.Queue()
+        self._tx_time_bias = 0.5
+        self._tx_time_dev = 0.3
+
         self._last_rssi = -147.0
         self._frequency_error = 123.45
 
+        self.debug = RadioDebug(self)
```

Comment:
I think self.debug is probably not a great name as it might get easily confused with the self.debug we use to log things to the console. Maybe `self.test_utils` or something along those lines.

Reply:
I agree.